sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java

```diff
@@ -284,7 +284,7 @@ CosmosException setDiagnostics(CosmosDiagnostics cosmosDiagnostics) {
     @Override
     public String toString() {
         return getClass().getSimpleName() + "{" + "userAgent=" + USER_AGENT + ", error=" + cosmosError + ", resourceAddress='"
-                   + resourceAddress  +  "', requestUri='" + (requestUri != null ? requestUri.getURIAsString() : null) +  "', statusCode=" + statusCode + ", message=" + getMessage()
+                   + resourceAddress  +  ", statusCode=" + statusCode + ", message=" + getMessage()
                    + ", causeInfo=" + causeInfo() + ", responseHeaders=" + responseHeaders + ", requestHeaders="
```

Comment:
why we remove requestUri?
And I think currently we are mixing using resourceAddress and requestUri:
for example in RxGatewayStoreModel, we are passing resourceId to resourceAddress (XwJnANUYRi0=)
But in RntbdRequestManager, we are passing physical address to resourceAddress

Reply:
Removed requestUri from CosmosException string representation, as it is never set. 
In RxGatewayStoreModel, we are passing `request.getResourceAddress()`

Reply:
This is how it looks for gateway -> resourceAddress='dbs/RxJava.SDKTest.SharedDatabase_20201109T182237_GSh/colls/3d2be199-e23b-4af2-a755-510d7068ff97/docs/18b9d280-e899-4300-ae0c-5fe56411aa18, 

And for direct -> resourceAddress='rntbd://cdb-ms-prod-westus2-fd26.documents.azure.com:14376/apps/db2a149b-bdb4-4755-b26e-56d1a86cdd4e/services/8ce44d24-fe68-49ae-9784-94b322fa46a3/partitions/003a20cb-a187-4b3a-9bc9-306be0217c36/replicas/132494067072942319s/, 

which both are resource addresses.

Reply:
Thanks Kushagra.
I just wonder should we also capture the server info for gateway mode: (like following)
https://127.0.0.1:8081/dbs/RxJava.SDKTest.SharedDatabase_20201109T193921_erZ/colls/7d5337a3-30f7-46bb-b318-104a5dcfb721

Currently, the resourceAddress for gateway mode only contains the resourceId or full name, but not a complete uri.

Reply:
Good point, not sure how we can capture that, any ideas ?

Reply:
Currently we calculate the uri at RxGatewayStoreModel.performRequest, maybe we can add it in the requestContext?

Reply:
Yes, I will try to do that, so we can re-use it, will update this thread once I have reached a conclusion :)

Reply:
what about the direct mode? isn't that set in the direct mode? @kushagraThapar  could you please check.

Reply:
@moderakh  - It is not used anywhere, other than at this code piece - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java#L808
Which I am not even sure, if that is correct or not, but don't plan to touch it  :)

Reply:
@xinlian12 @FabianMeiswinkel  - I have added the full resource address in gateway mode too, please check. 
Also updated the PR with sample.

Reply:
Thanks Kushagra, looks good to me~